packages/gatsby-telemetry/src/telemetry.js

```diff
@@ -0,0 +1,196 @@
+  store = new EventStorage()
+  debouncer = {}
+  metadataCache = {}
+  defaultTags = {}
+  osInfo // lazy
+  trackingEnabled // lazy
+  componentVersion
+  sessionId
+  constructor() {
+    this.sessionId = uuid()
+    try {
+      this.componentVersion = require(`../package.json`).version
+    } catch (e) {
+      // ignore
+    }
+  }
+
+  captureEvent(type = ``, tags = {}) {
+    if (!this.isTrackingEnabled()) {
+      return
+    }
+    if (Array.isArray(type) && type.length > 2) {
+      type = type[2].toUpperCase()
+    }
+    const decoration = this.metadataCache[type]
+    delete this.metadataCache[type]
+    const eventType = `CLI_COMMAND_${type}`
+    this.buildAndStoreEvent(eventType, Object.assign(tags, decoration))
+  }
+
+  captureError(type, tags = {}) {
+    if (!this.isTrackingEnabled()) {
+      return
+    }
+    const decoration = this.metadataCache[type]
+    delete this.metadataCache[type]
+    const eventType = `CLI_ERROR_${type}`
+    sanitizeError(tags)
+
+    // JSON.stringify won't work for Errors w/o some trickery:
+    let { error } = tags
+    if (error) {
+      error = error.map(e =>
+        JSON.parse(JSON.stringify(e, Object.getOwnPropertyNames(e)))
+      )
+    }
+    tags.error = JSON.stringify(error)
+
+    this.buildAndStoreEvent(eventType, Object.assign(tags, decoration))
+  }
+
+  captureBuildError(type, tags = {}) {
+    if (!this.isTrackingEnabled()) {
+      return
+    }
+    const decoration = this.metadataCache[type]
+    delete this.metadataCache[type]
+    const eventType = `BUILD_ERROR_${type}`
+    sanitizeError(tags)
+    tags.error = JSON.stringify(tags.error)
+
+    this.buildAndStoreEvent(eventType, Object.assign(tags, decoration))
+  }
+
+  buildAndStoreEvent(eventType, tags) {
+    const event = {
+      ...this.defaultTags,
+      ...tags, // The schema must include these
+      eventType,
+      sessionId: this.sessionId,
+      time: new Date(),
+      machineId: this.getMachineId(),
+      repositoryId: this.getRepoId(),
+      componentId: `gatsby-cli`,
+      osInformation: this.getOsInfo(),
+      componentVersion: this.componentVersion,
+    }
+    this.store.addEvent(event)
+  }
+
+  getMachineId() {
+    // Cache the result
+    if (this.machineId) {
+      return this.machineId
+    }
+    let machineId = this.store.getConfig(`machineId`)
+    if (!machineId) {
+      machineId = uuid()
+      this.store.updateConfig({ machineId })
+    }
+    this.machineId = machineId
+    return machineId
+  }
+
+  isTrackingEnabled() {
+    // Cache the result
+    if (this.trackingEnabled !== undefined) {
+      return this.trackingEnabled
+    }
+    let telemetry = this.store.getConfig(`telemetry`)
+    if (telemetry === undefined || telemetry === null) {
+      console.log(
+        `Gatsby has started collecting anonymous usage analytics to help improve Gatsby for all users.\n` +
+          `If you'd like to opt-out, you can use \`gatsby analytics --disable\`\n` +
```

Comment:
```suggestion
          `If you'd like to opt-out, you can use \`gatsby telemetry --disable\`\n` +
```

Reply:
Also this command will fail if user didn't update global `gatsby-cli`, so troubleshooting step (i.e. "if command is not recognized, update "gatsby-cli" with `npm i -g gatsby-cli` command")